fix: Component notes do not work when using twigAdapter

rchavik opened the pull request:
Closes: #45

mihkeleidast commented on src/adapter.js:
```diff
@@ -143,6 +143,10 @@ class TwigAdapter extends Fractal.Adapter {
         return new Promise(function(resolve, reject){
 
+            if (path == null) {
+                return resolve(str);
```
returning early means that the notes are not passed through the twig engine at all, which is also unexpected.

i would remove that if and on the following line, conditionally use `Path.relative`:
```js
let tplPath = path ? Path.relative(self._source.fullPath, path) : undefined;
```

That would, in my testing, make the most of the situation. For example, that will allow users to use the global data that is passed into context in the notes file:
```
## config
<ul>
{% for key, value in _config %}
  <li>{{ key }}: {{ value }}</li>
{% endfor %}
</ul>
```

rchavik replied:
You're right. I have updated the PR.

rchavik committed:
fix: Component notes do not work when using twigAdapter

Adjusted to use better fix from: https://github.com/frctl/twig/pull/46#pullrequestreview-386707440

Closes: #45
